src/Models/LocationPaymentSetting.php

```diff
@@ -28,4 +48,33 @@ public function location()
     {
         return $this->belongsTo(Location::class);
     }
+
+    public function getStripePublishableAttribute($value)
+    {
+        return $value ?? $this->getStripeConfigKey('publishable');
+    }
+
+    public function getStripeSecretAttribute($value)
+    {
+        return $value ?? $this->getStripeConfigKey('secret');
+    }
```

Comment:
These attribute getters (and the `getStripeConfigKey()` helper) is the key logic.  If a DB value is set for either of these attributes, its used.  Otherwise, it goes to the config.  Within the config, it searches for a location specific entry (keyed by location abbreviation).  If not configured, then a `default` key within the config is used.  If that's not configured, then there are no stripe keys available and null is returned.

Reply:
This is an excellent solution!